client/asset/dcr/config.go

```diff
@@ -30,30 +30,24 @@ var (
 	defaultConfigPath = filepath.Join(dcrwHomeDir, "dcrwallet.conf")
 )
 
-// DCRConfig is passed to the constructor.
-type DCRConfig struct {
-	// RPCUser is the RPC username provided to dcrwallet configuration.
-	RPCUser string `ini:"username"`
-	// RPCPass is the RPC password provided to dcrwallet configuration.
-	RPCPass string `ini:"password"`
-	// RPCListen is the RPC network address provided to dcrwallet configuration.
-	// If the value is an empty string, it will be set to a default value for the
-	// network.
-	RPCListen string `ini:"rpclisten"`
-	// RPCCert is the filepath to the dcrwallet TLS certificate. If it is not
-	// provided, the default dcrwallet location will be assumed.
-	RPCCert string `ini:"rpccert"`
+// Config holds the parameters needed to initialize an RPC connection to a dcr
+// wallet. Default values are used for RPCListen and/or RPCCert if not set.
+type Config struct {
+	RPCUser   string `ini:"username, RPC Username, Username for RPC connections"`
+	RPCPass   string `ini:"password, RPC Password, Password for RPC connections"`
+	RPCListen string `ini:"rpclisten, RPC Address (host or host:port), dcrwallet interface/port for RPC connections (default port: 9109, testnet: 19109)"`
+	RPCCert   string `ini:"rpccert, TLS Certificate, Path to the dcrwallet TLS certificate file"`
```

Comment:
While implementing the form on the webserver frontend, I realized that this needs to be the file contents (aka Text) rather than the path to the file primarily because a user may be running the webserver frontend on a machine different from the webserver host machine, and a file present on the user's machine at some path would not necessarily be present on the same path on the server or hold the same content.

I think we should declare that out of scope for this PR but ultimately, we should record the "data type" of each config option so that the appropriate data collection widget may be used on the frontend  and some client-side data validation performed where applicable. For e.g. what fields are optional vs what fields are required? `Ports` should obv be numbers, `Certs` should be files (actually `LongText` read from files) etc.

Also, when considering that the user-provided rpc cert path may be non-existent on the server, though present on the user's computer, the same considerations would apply to providing config info via config file upload on the frontend. An uploaded config file would naturally define a locally available rpc cert path, which may not be found on the server. Perhaps instead of expecting a path in the config, we expect the rpc cert text?

Can make this into an issue to be looked into sometime later.

Reply:
Good observation. User's will just need to specify the filepath on the `dexc` machine for now. Please do open an issue if you have time.

A proper solution would likely involve a more complex `config.Option` structure that can't be done with just the `ini` struct tag.